src/services/CertificateGenerationService.ts

```diff
@@ -114,35 +131,54 @@ class CertificateGenerationService {
 	}
 
 	/**
-	 * Handler method for retrieving feature flags and checking if test station is in Wales
+	 * Retrieve feature flags by using or setting cache
+	 */
+	public async retrieveAndSetFlags() {
+		if (CertificateGenerationService.flags) {
+			console.log('Feature flag cache already set');
+			return;
+		} else {
```

Comment:
Once again not a blocker for this PR, but if we are bailing out early, we don't need the else, we can remove the nesting/wrapping the `else` creates e.g.

```
public async retrieveAndSetFlags() {
		if (CertificateGenerationService.flags) {
			console.log('Feature flag cache already set');
			return;
		}

		console.log('Feature flag cache not set, retrieving feature flags');
		
		try {
		    CertificateGenerationService.flags = await getProfile();
		} catch (e) {
		    console.error(`Failed to retrieve feature flags - ${e}`);
		}	

		console.log('Using feature flags ', CertificateGenerationService.flags);
	}
```

Reply:
Same here, updated removing obsolete else and pushed.